app/master/slave.py

```diff
@@ -48,25 +47,24 @@ def mark_as_idle(self):
 
         self.current_build_id = None
 
-    def setup(self, build_id, project_type_params, build_executor_start_index):
+    def setup(self, build):
```

Comment:
Yay, so much nicer. I think originally we (I?) were too worried about making law of demeter violations, but this is better and makes sense.